opslevel/resource_opslevel_check_repository_integrated.go

```diff
@@ -1,59 +1,196 @@
-// }
-
-// func resourceCheckRepositoryIntegratedRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	id := d.Id()
-
-// 	resource, err := client.GetCheck(opslevel.ID(id))
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	if err := setCheckData(d, resource); err != nil {
-// 		return err
-// 	}
-
-// 	return nil
-// }
-
-// func resourceCheckRepositoryIntegratedUpdate(d *schema.ResourceData, client *opslevel.Client) error {
-// 	checkUpdateInput := getCheckUpdateInputFrom(d)
-// 	input := opslevel.NewCheckUpdateInputTypeOf[opslevel.CheckRepositoryIntegratedUpdateInput](checkUpdateInput)
-
-// 	if _, err := client.UpdateCheckRepositoryIntegrated(*input); err != nil {
-// 		return err
-// 	}
-// 	d.Set("last_updated", timeLastUpdated())
-// 	return resourceCheckRepositoryIntegratedRead(d, client)
-// }
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+	"github.com/relvacode/iso8601"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &CheckRepositoryIntegratedResource{}
+	_ resource.ResourceWithImportState = &CheckRepositoryIntegratedResource{}
+)
+
+func NewCheckRepositoryIntegratedResource() resource.Resource {
+	return &CheckRepositoryIntegratedResource{}
+}
+
+// CheckRepositoryIntegratedResource defines the resource implementation.
+type CheckRepositoryIntegratedResource struct {
+	CommonResourceClient
+}
+
+type CheckRepositoryIntegratedResourceModel struct {
+	Category    types.String `tfsdk:"category"`
+	Description types.String `tfsdk:"description"`
+	Enabled     types.Bool   `tfsdk:"enabled"`
+	EnableOn    types.String `tfsdk:"enable_on"`
+	Filter      types.String `tfsdk:"filter"`
+	Id          types.String `tfsdk:"id"`
+	Level       types.String `tfsdk:"level"`
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+}
+
+func NewCheckRepositoryIntegratedResourceModel(ctx context.Context, check opslevel.Check) CheckRepositoryIntegratedResourceModel {
+	var model CheckRepositoryIntegratedResourceModel
+
+	model.Category = types.StringValue(string(check.Category.Id))
+	model.Enabled = types.BoolValue(check.Enabled)
+	model.EnableOn = types.StringValue(check.EnableOn.Time.Format(time.RFC3339))
+	model.Filter = types.StringValue(string(check.Filter.Id))
+	model.Id = types.StringValue(string(check.Id))
+	model.Level = types.StringValue(string(check.Level.Id))
+	model.Name = types.StringValue(check.Name)
+	model.Notes = types.StringValue(check.Notes)
+	model.Owner = types.StringValue(string(check.Owner.Team.Id))
+	model.LastUpdated = timeLastUpdated()
```

Comment:
Needs to be moved

Reply:
Also, we should use the terraform type conversion functions in the `NewResourceModel` functions here as part of standardization -- non-blocking.

Reply:
done